vault/logical_system.go

```diff
@@ -2922,8 +2922,8 @@ func (b *SystemBackend) handleMonitor(ctx context.Context, req *logical.Request,
 	if logLevel == log.NoLevel {
 		return logical.ErrorResponse("unknown log level"), nil
 	}
-
-	flusher, ok := w.ResponseWriter.(http.Flusher)
+	nw := w.ResponseWriter.(*logical.NamespaceResponseWriter)
```

Comment:
Are you sure this is necessary (genuine question, I'm not saying it isn't)?  Since we're embedding the ResponseWriter without giving it an explicit field name, I would've thought we could treat our wrapping type NamespaceResponseWriter more or less as a drop-in replacement.

Reply:
So, I am getting this error with the old code:
"""
--- FAIL: TestSysMonitorStreamingLogs (3.61s)
    /Users/hamid/repo/vault/http/sys_monitor_test.go:57: Error making API request.
        
        URL: GET https://127.0.0.1:62014/v1/sys/monitor?log_level=DEBUG
        Code: 400. Errors:
        
        * streaming not supported
"""        

and flusher is nil. 

The http.Flusher comments say this: 

"""
The default HTTP/1.x and HTTP/2 ResponseWriter implementations support Flusher, but ResponseWriter wrappers may not. Handlers should always test for this ability at runtime.
"""

I think the comment is talking about our use case as far as I understand.

Reply:
Ok, I stand corrected!  I'd feel a little better if you made the cast conditional though.  I know that with the current code it's always the same underlying type, but if things change and that's not always true, this would panic.

Reply:
Ah, I see. Sure thing, would this be fine ? 
"""
flusher, ok := w.ResponseWriter.(http.Flusher)
if !ok {
        // Casting to logical.ResponseWriter and try http.Flusher again
        nw := w.ResponseWriter.(*logical.NamespaceResponseWriter)
        flusher, ok = nw.ResponseWriter.(http.Flusher)
        if !ok {
	        return logical.ErrorResponse("streaming not supported"), nil
        }
}
"""

Reply:
Yeah I had in mind something like that.

Reply:
Great! Thanks